e2e/cypress/integration/multi_team_and_dm/close_current_dm_redirects_spec.js

```diff
@@ -0,0 +1,154 @@
+
+        // * Expect to be redirected to off-topic channel, check channel title and url
+        expectActiveChannelToBe('Off-Topic', offTopicChannelUrl);
+
+        // # From the 'Direct Messages' menu, find the same user as before and send 'hi'
+        sendDirectMessageToUser(firstDMUser, 'hi again');
+
+        // # Open the dirrect message's menu from the upper left section
+        cy.get('#channelHeaderDropdownButton').click();
+
+        // # Click on 'Close direct message' menu item
+        cy.get('#channelCloseMessage').click();
+
+        // * Expect to be redirected to off-topic channel, check channel title and url
+        expectActiveChannelToBe('Off-Topic', offTopicChannelUrl);
+    });
+
+    it('MM-T453 Closing a different direct message should not affect active direct message', () => {
+        // # Login as test user
+        cy.apiLogin(testUser);
+
+        // # View 'off-topic' channel
+        cy.visit(offTopicChannelUrl);
+
+        // # Send a direct message to a first user
+        sendDirectMessageToUser(firstDMUser, 'hi first');
+
+        // # Send a message to a second user
+        sendDirectMessageToUser(secondDMUser, 'hi second');
+
+        // # Close the direct message previously opened with the first user
+        closeDirectMessageViaXButton(testUser, firstDMUser, testTeam);
+
+        // * Expect channel title and url to secondDMUser's username
+        expectActiveChannelToBe(secondDMUser.username, `/messages/@${secondDMUser.username}`);
+    });
+
+    it('MM-T453 Changing URL to root url when viewing a direct message should redirect to direct message', () => {
+        // # Login as test user
+        cy.apiLogin(testUser);
+
+        // # View 'off-topic' channel
+        cy.visit(offTopicChannelUrl);
+
+        // # Send a direct message to a first user
+        sendDirectMessageToUser(firstDMUser, 'hi');
+
+        // # Visit root url
+        cy.visit('/');
+
+        // * Expect channel title and url to firstDMUser's username
+        expectActiveChannelToBe(firstDMUser.username, `/messages/@${firstDMUser.username}`);
+    });
+});
+
+const expectActiveChannelToBe = (title, url) => {
+    // * Expect channel title to match title passed in argument
+    cy.get('#channelHeaderTitle').
+        should('be.visible').
+        and('contain.text', title);
+
+    // * Expect url to match url passed in argument
+    cy.url().should('contain', url);
+};
+
+const sendDirectMessageToUser = (user, message) => {
+    // # Open a new direct message with firstDMUser
+    cy.get('#addDirectChannel').click();
+
+    // # Type username
+    cy.get('#selectItems input').should('be.enabled').type(`@${user.username}`, {force: true});
+
+    // * Expect user count in the list to be 1
+    cy.get('#multiSelectList').
+        should('be.visible').
+        children().
+        should('have.length', 1);
+
+    // # Select first user in the list
+    cy.get('body').
```

Comment:
I have to use the keyboard here because everytime I try to click on the item, even after asserting its visibility, it fails because Cypress thinks it's detached from the DOM